docs/tutorial/index.rst

```diff
@@ -1,4 +1,147 @@
+    ...           Event(name='Talk 11', event_type='talk', duration=30, roles="empty", tags=['advanced'], unavailability=outside_slots, demand=30),
+    ...           Event(name='Talk 12', event_type='talk', duration=30, roles="empty", tags=['advanced'], unavailability=outside_slots, demand=30),
+    ...           Event(name='Workshop 1', event_type='workshop', duration=60, roles="empty", tags=['testing'], unavailability=talk_slots + outside_slots, demand=300),
+    ...           Event(name='Workshop 2', event_type='workshop', duration=60, roles="empty", tags=['testing'], unavailability=talk_slots + outside_slots, demand=40),
+    ...           Event(name='City tour', event_type='social event', duration=90, roles="empty", tags=[], unavailability=talk_slots + workshop_slots, demand=100),
+    ...           Event(name='Boardgames', event_type='social event', duration=90, roles="empty", tags=[], unavailability=talk_slots + workshop_slots, demand=20)]
+
+Further to this we have a couple of other constraints:
+
+
+- The speaker for :code:`Talk 1` is also the person delivering :code:`Workshop 1`::
+
+        >>> events[0].unavailability.extend([events[6]])
+
+- Also, the person running :code:`Workshop 2` is the person hosting the
+  :code:`Boardgames`::
+
+        >>> events[13].unavailability.append(events[-1])
+
+Now that we have :code:`slots` and :code:`events` we can easily schedule our
+event::
+
+    >>> from conference_scheduler import scheduler
+    >>> schedule = scheduler.schedule(events, slots)
+
+This schedule is a generator::
+
+    >>> schedule = sorted(schedule, key=lambda item: item.slot.starts_at)
+    >>> for item in schedule:
+    ...     print(f"{item.event.name} at {item.slot.starts_at} in {item.slot.room.name}")
+    Talk 3 at 15-Sep-2016 09:30 in Small
+    Talk 11 at 15-Sep-2016 09:30 in Big
+    Talk 4 at 15-Sep-2016 10:00 in Small
+    Talk 10 at 15-Sep-2016 10:00 in Big
+    Talk 5 at 15-Sep-2016 12:30 in Big
+    Talk 9 at 15-Sep-2016 12:30 in Small
+    Talk 6 at 15-Sep-2016 13:00 in Big
+    Talk 8 at 15-Sep-2016 13:00 in Small
+    Talk 1 at 16-Sep-2016 09:30 in Big
+    Workshop 2 at 16-Sep-2016 09:30 in Small
+    Talk 2 at 16-Sep-2016 10:00 in Big
+    Talk 7 at 16-Sep-2016 12:30 in Big
+    City tour at 16-Sep-2016 12:30 in outside
+    Talk 12 at 16-Sep-2016 13:00 in Big
+    Workshop 1 at 16-Sep-2016 13:00 in Small
+    Boardgames at 16-Sep-2016 13:00 in outside
+
+
+
+We see that all the events are scheduled in appropriate rooms (as indicated by
+the unavailability attribute for the events). Also we have that :code:`Talk 1`
+doesn't class with :code:`Workshop 1`.
+Similarly, the :code:`Boardgame` does not clash with :code:`Workshop 2`.
+
+You will also note that in any given session, talks share at least one tag. This
+is another constraint of the model, if you find that your schedule has no
+solutions you can adjust it by recategorising your talks (or giving them all a
+single category).
```

Comment:
I feel like maybe I’ve misunderstood something here? Seems like talks in the same session don’t share a tag – and that’s what you’d want, if I’ve understood tags correctly?

e.g. the 9:30 slot has Talk 3 and Talk 11, which have "beginner" and "advanced" tags.